tests/unit/test_charm.py

```diff
@@ -513,12 +509,188 @@ def test_charm_writes_meaningful_alerts_filename_4(self, *_):
+                # Nothing else to do here.)
+
+                # AND reload is not invoked
+                reload_config_patch.assert_not_called()
+
+    @k8s_resource_multipatch
+    @patch("lightkube.core.client.GenericSyncClient")
+    @patch("socket.getfqdn", new=lambda *args: "fqdn")
+    @patch("ops.testing._TestingPebbleClient.replan_services")
+    @patch("ops.testing._TestingPebbleClient.start_services")
+    @patch("ops.testing._TestingPebbleClient.restart_services")
+    @patch("prometheus_server.Prometheus.reload_configuration")
+    def test_workload_restarts_when_some_config_options_change(
+        self, reload_config, restart, start, replan, *_
+    ):
+        """Some config options go in as cli args and require workload restart."""
+        # GIVEN a pebble plan
+        first_plan = self.plan
+        self.assertTrue(self.service.is_running())
+
+        # WHEN web_external_url is set
+        self.harness.update_config({"web_external_url": "http://test:80/foo/bar"})
+
+        # THEN pebble service is updated
+        second_plan = self.plan
+        self.assertEqual(cli_arg(second_plan, "--web.external-url"), "http://test:80/foo/bar")
+        self.assertNotEqual(first_plan.to_dict(), second_plan.to_dict())
+
+        # AND workload is restarted
+        self.assertTrue(self.service.is_running())
+        self.assertTrue(restart.called or start.called or replan.called)
+        restart.reset_mock()
+        start.reset_mock()
+        replan.reset_mock()
+
+        # BUT reload is not invoked
+        reload_config.assert_not_called()
+
+        # WHEN web_external_url is changed
+        self.harness.update_config({"web_external_url": "http://test:80/foo/bar/baz"})
+
+        # THEN pebble service is updated
+        third_plan = self.plan
+        self.assertEqual(cli_arg(third_plan, "--web.external-url"), "http://test:80/foo/bar/baz")
+        self.assertNotEqual(second_plan.to_dict(), third_plan.to_dict())
+
+        # AND workload is restarted
+        self.assertTrue(self.service.is_running())
+        self.assertTrue(restart.called or start.called or replan.called)
+        restart.reset_mock()
+        start.reset_mock()
+        replan.reset_mock()
+
+        # BUT reload is not invoked
+        reload_config.assert_not_called()
+
+        # WHEN web_external_url is unset
+        self.harness.update_config(unset=["web_external_url"])
+
+        # THEN pebble service is updated
+        fourth_plan = self.plan
+        self.assertEqual(cli_arg(fourth_plan, "--web.external-url"), "http://fqdn:9090")
+        self.assertNotEqual(third_plan.to_dict(), fourth_plan.to_dict())
+
+        # AND workload is restarted
+        self.assertTrue(self.service.is_running())
+        self.assertTrue(restart.called or start.called or replan.called)
+        restart.reset_mock()
+        start.reset_mock()
+        replan.reset_mock()
+
+        # BUT reload is not invoked
+        reload_config.assert_not_called()
```

Comment:
Maybe I'm missing something, but aren't all of these essentially checking the same behavior?

Reply:
Yes, knowing how the charm is written then these are redundant.
OTOH, feature-wise, the test goes through the "lifecycle" of a config option: set-change-unset. We want to make sure this promise is kept in future changes.